iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -91,6 +111,66 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task GetWritablePropertiesAndHandleChangesAsync()
+        {
+            ClientProperties properties = await _deviceClient.GetClientPropertiesAsync();
+            ClientPropertyCollection writableProperties = properties.WritablePropertyRequests;
+            long serverWritableProperties = writableProperties.Version;
+
+            // Check if the writable property version is outdated on the local side.
+            // For the purpose of this sample, we'll only check the writable property versions between local and server
+            // side without comparing the property values.
+            if (serverWritableProperties > s_localWritableProperties)
+            {
+                _logger.LogDebug($"The writable property version cached on local is changing from {s_localWritableProperties} to {serverWritableProperties}.");
+
+                foreach (KeyValuePair<string, object> writableProperty in writableProperties)
+                {
+                    switch (writableProperty.Key)
+                    {
+                        case Thermostat1:
+                        case Thermostat2:
+                            const string targetTemperatureProperty = "targetTemperature";
+                            string componentName = writableProperty.Key;
+
+                            if (writableProperties.TryGetValue(componentName, targetTemperatureProperty, out double targetTemperatureValue))
+                            { 
+                                _logger.LogDebug($"Property: Received - component=\"{componentName}\", [ \"{targetTemperatureProperty}\": {targetTemperatureValue}°C ].");
+
+                                _temperature[componentName] = targetTemperatureValue;
+
+                                string propertyValueString = $"{{ \"value\": {_temperature[componentName]}, \"ac\": {CommonClientResponseCodes.OK}, \"av\": {serverWritableProperties} }}";
+                                var propertyValue = SystemTextJsonPayloadSerializer.Instance.DeserializeToType<object>(propertyValueString);

```

Comment:
This can be replaced by using `SystemTextJsonPayloadSerializer.Instance.CreateWritablePropertyResponse` to create the response in the expected format (this will put it in the ac/av/ad/value format)

Reply:
same for the thermostat sample as well

Reply:
Should end users be referencing `SysteTextJsonPayloadSerializer.Instance` directly from code that isn't initializing the client options?

Reply:
They can, yes. It is public. For the property update requests that the SDK automatically handles for you, we provide the `WritableClientProperty` that encapsulates this logic. For property update requests that the client could have "missed" and is required to retrieve from the `ClientPropertyCollection` directly, they'll need to do the transformation to the expected reported property format themselves.
At this point, instead of getting into json string formatting, they can make use of `SystemTextJsonPayloadSerializer.Instance.CreateWritablePropertyResponse`

Reply:
Good points! Modified.